plugins/catalog-react/src/components/EntityOwnerPicker/useFacetsEntities.ts

```diff
@@ -41,40 +41,63 @@ type FacetsInitialRequest = {
  * hook, which is also used by EntityOwnerPicker.
  * In this mode, the EntityOwnerPicker won't show detailed information of the owners.
  */
-export function useFacetsEntities({ enabled }: { enabled: boolean }) {
+export function useFacetsEntities({
+  enabled,
+  selectedEntityKind,
+}: {
+  enabled: boolean;
+  selectedEntityKind?: string;
+}) {
   const catalogApi = useApi(catalogApiRef);
+  const [facetsPromise, setFacetsPromise] = useState<Promise<Entity[]>>(
+    Promise.resolve([]),
+  );
 
-  const [facetsPromise] = useState(async () => {
-    if (!enabled) {
+  const fetchFacetsEntities = useCallback(async (): Promise<Entity[]> => {
```

Comment:
It feels slightly convoluted now with a `useCallback` and `useEffect`; could this be a single `useMemo` instead that depends on `[catalogApi, selectedEntityKind, enabled]`? Which evaluates to a `Promise<Entity[]>` still, of course.